src/messageComposer/middleware/textComposer/commands.ts

```diff
@@ -142,29 +148,49 @@ export const createCommandsMiddleware = (
           return next(newState);
         }
 
+        const query = triggerWithToken.slice(1);
+
+        const searchQuery = getFirstWordFromText(query);
+
+        const commands = searchSource?.query(searchQuery).items;
+
+        const matchedCommand = commands?.find((command) =>
```

Comment:
@khushal87 does it make sense to perform query(searchQuery) and then again commands.find()? Isn't it doing the same with the difference that commands.find in this case would just be commands[0] (as find returns on the first matched and the first matched is the at 0 index before query returns all the matches)?

Reply:
I will simplify it. Makes sense

Reply:
Done